GliaWidgets/SecureConversations/SecureConversations.FileUploadList.swift

```diff
@@ -0,0 +1,965 @@
+            contentView.autoMatch(.width, to: .width, of: scrollView)
+
+            contentView.addSubview(stackView)
+            stackView.autoPinEdgesToSuperviewEdges()
+        }
+
+        func updateHeight() {
+            let height = uploadViews
+                .prefix(props.maxUnscrollableViews)
+                .reduce(CGFloat.zero) { result, view in
+                    result + view.systemLayoutSizeFitting(UIView.layoutFittingCompressedSize).height
+                }
+
+            heightLayoutConstraint.constant = height
+        }
+
+        /// Make `fileUploadView` and `fileUploadView.removeButton` "visible"
+        /// for VoiceOver.
+        func addAccessibilityProperties(for fileUploadView: FileUploadView) {
+            stackView.accessibilityElements?.append(fileUploadView)
+            stackView.accessibilityElements?.append(fileUploadView.removeButton)
+        }
+
+        /// Remove fileUploadView` and `fileUploadView.removeButton` from
+        /// `accessibilityElements`.
+        func removeAccessibilityProperties(for fileUploadView: FileUploadView) {
+            stackView.accessibilityElements?.removeAll(
+                where: {
+                    guard let view = $0 as? UIView else { return false }
+                    return view === fileUploadView || view === fileUploadView.removeButton
+                }
+            )
+        }
+
+        /// Clear all `accessibilityElements`at once.
+        func removeAccessibilityPropertiesForAllUploadViews() {
+            stackView.accessibilityElements?.removeAll()
+        }
+
+        func renderProps() {
+            // Since we need to create/remove views based on the diff
+            // in collection of props, we need to have association
+            // between each element in collection of props and each
+            // corresponding view. This is achieved by having common
+            // id for props and view.
+
+            // Check for removal of views.
+            var toRemove = [FileUploadView.Props.Identifier]()
+            for id in cachedViews.ids where props.uploads[by: id] == nil {
+                toRemove.append(id)
+            }
+
+            for id in toRemove {
+                guard let view = cachedViews[by: id] else { continue }
+                removeUploadView(view)
+                cachedViews.remove(by: id)
+            }
+
+            // Check for addition of views.
+            for id in props.uploads.ids where cachedViews[by: id] == nil {
+                let uploadView = FileUploadView()
+                addUploadView(uploadView)
+                cachedViews.append(item: uploadView, identified: id)
+            }
+
+            // Assign props for corresponding views
+            for uploadProps in props.uploads {
+                cachedViews[by: uploadProps.id]?.props = uploadProps
+            }
+        }
+    }
+}
+
+extension SecureConversations.FileUploadListView {
+    class FileUploadView: UIView {
```

Comment:
I'd like to propose
- make this class final
- inherit from BaseView
- move to separate file